DTS-STN/Scrum-Poker
# Non functional card selector for home page

## [DUMS-95- Client side card selector before the user creates a room](https://jira-dev.bdm-dev.dts-stn.com/browse/DUMS-95) (Jira Issue)

### Description

This PR adds a card selector to the home page, with default values selected as per Figma designs. When a user selects (or unselects) a card from the list, the value is either added to or removed from the `cardList` array (which will later be used to update the room upon creation). I've chosen to stay away from `useState` for updating the card list since the value was only getting added after state had been updated, which is not ideal. Instead I used `useRef` which updates the array in real time, but since this is not communicated to the browser, there is no re-render and as such we run into problems with rendering the `selected` styles on the cards when they are clicked. Adding the `forceUpdate` reducer and calling it whenever a card is clicked fixes this, but I am open to suggestions if this isn't ideal.

### What to test for/How to test

1. Pull in branch
2. Type `npm run dev`
3. Enter a name and unselect all the cards
4. Click create room, you should be shown an error asking you to select at least one card
5. Select a card and click create room again, you should be redirected to a room (the cards you selected won't be the ones shown in the room as this functionality isn't in this PR)

### Additional Notes

I've also moved the array of card SVGs into a separate file that can be imported instead of doubling up on code on the home and room pages. I'll make use of this when I have the room communicate with the server to display the selected cards.
